grin/test/LintSpec.hs

```diff
@@ -224,45 +224,38 @@ spec = do
       let (_, errors) = lint allWarnings (Just typeEnv) program
       lintErrors errors `shouldBe` ["Invalid pattern match for (CInt x). Expected pattern of type: {CInt[T_Dead]}, but got: {CFloat[T_Float]}"]
 
-    it "disregards variable patterns" $ do
+    it "doesn't alert over-approximated binds" $ do
       let program = [prog|
           main =
-            n0 <- pure (CInt 0)
-            n1 <- case n0 of
-              (CInt c0) -> pure n0
-              (CFloat c1) ->
-                a0 <- pure (CFloat 2.0)
-                pure a0
+            i <- pure (CInt 1)
+            f <- pure (CFloat 1.0)
+            l <- store i
+            update l f
+            v <- fetch l
+            (CFloat f2) <- pure v
             pure ()
         |]
       let typeEnv = inferTypeEnv program
       let (_, errors) = lint allWarnings (Just typeEnv) program
       lintErrors errors `shouldBe` []
 
-    -- NOTE: Bottom-up typing can only approximate the result of HPT.
-    it "can give false positive errors" $ do
+    it "disregards variable patterns" $ do
       let program = [prog|
           main =
-            n0 <- case 0 of
-              0 ->
-                n1 <- pure (CInt 0)
-                pure n1
-              1 ->
-                n2 <- pure (CFloat 0.0)
-                pure n2
-            (CInt x) <- case n0 of
+            n0 <- pure (CInt 0)
+            n1 <- case n0 of
               (CInt c0) -> pure n0
               (CFloat c1) ->
-                a0 <- pure (CInt 0)
+                a0 <- pure (CFloat 2.0)
                 pure a0
             pure ()
         |]
```

Comment:
Could you please name all subexpressions, so that adopting these tests to the new syntax will be easier?